src/modules/dataGroups.js

```diff
@@ -37,40 +35,47 @@ class DataGroups extends React.PureComponent {
     }
 
     resizeWindow() {
-        const { columns } = this.props;
-        const columnLength = columns.length;
-        let columnCount = Math.floor(this.dataGroups.offsetWidth / 298);
-
-        if (columnCount < 1) {
-            columnCount = 1;
-        } else if (columnCount > columnLength) {
-            columnCount = columnLength;
-        }
-
-        this.reorderColumns(columnCount);
+        this.reorderColumns();
     }
 
-    reorderColumns(columnCount) {
+    reorderColumns() {
         const { columns } = this.props;
-        const newColumns = [];
+        const blockSize = this.dataGroups.offsetWidth;
+        const columnMinWidth = 300;
+        const columnsLength = columns.length;
+        let amountOfColumns = Math.floor(blockSize / columnMinWidth);
+        let newColumns = [];
+
+        if (amountOfColumns === 1 || amountOfColumns < 1) {
+            amountOfColumns = 1;
+        } else if (amountOfColumns > columnsLength) {
+            amountOfColumns = columnsLength;
+        }
+
         let columnNum = 0;
 
-        while (columnNum < columnCount) {
-            for (let i = 0; i < columns.length; i += columnCount) {
+        while (columnNum < amountOfColumns) {
+            for (let i = 0; i < columns.length; i += amountOfColumns) {
                 const column = columns[i + columnNum];
 
                 if (column !== undefined) {
-                    newColumns.push(column);
```

Comment:
`push()` is not wrong here per se, since `newColumns` is a local variable, so we don't have to avoid mutating it.  It's fine the way you have it as well, but just letting you know.  AFAIK, immutability is mostly important for shared values, i.e. from component state, Redux state or Flux state.  But it is always proper to mutate a local variable within the scope of a function body.